src/lib/common/globals.cpp

```diff
@@ -57,6 +57,7 @@ bool                   notifQueueStatistics = false;
 bool                   checkIdv1            = false;
 unsigned long long     inReqPayloadMaxSize  = DEFAULT_IN_REQ_PAYLOAD_MAX_SIZE;
 unsigned long long     outReqMsgMaxSize     = DEFAULT_OUT_REQ_MSG_MAX_SIZE;
+bool                   onlyMetadataChange   = false;
```

Comment:
This was already commented in the original PR https://github.com/telefonicaid/fiware-orion/pull/4274 (sorry I haven't spotted until now... a lot of file changes in this PR :)

The same comment stills: we don't want this to be a globally defined behaviour, but a per-subscription behaviour.

In fact, it seems that you have implemented all the CRUD for this attribute in the subscription lifecycle, so probably this is a remain of the old PR that can be safely removed.

Reply:
It is little bit painful, but we can see there are several global variables already present can we go with it. does it make any impact on the current application.

Reply:
Painful or not :) note that different subscription may have different settings. Subscription A may use onlyMetadataChange true and subscription B may use onlyMetadataChange false. If Orion process an update entity request triggering A and an update request triggering B at the same time then you are in problems if your solution is based in a global variable.

If this is complex and time consuming for you, let's do what I proposed in the other comment:

> If you find that difficult I could try to provide some help (the most "heavy" part of the work, about CRUD management of the new attribute, the new .test and the changes in existing .test files is already done).

Please tell me if you agree with that, so I can merge this PR in a pre-landing branch and you can continue working on Orion in a next issue (specially the ones labeled with [help wanted](https://github.com/telefonicaid/fiware-orion/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22)

Reply:
Thanks for your response.
Yes, I agree with you in case of per subscriptions global variable will definitely create problem. but I was thinking this feature as a static configuration for all subscriptions. The understanding was not very clear to me earlier but now it's pretty clear.
I will try to continue on this issue only.

Reply:
This is getting somehow urgent for us (we need to use it in some project) so I will take care of it.

We are going to merge this PR in a pre-landing branch and I'll continue with the final fixes. Thanks @chetan-NEC for your work! You can consider it as finished :)

CC: @hervaste

Reply:
Thanks for your response.

I'll start working on the next issue now.